src/communication/index.ts

```diff
@@ -83,29 +66,33 @@ export async function initLibp2p (): Promise<Libp2p> {
   return createLibP2P({ ...libp2pConf, peerId: await PeerId.create() })
 }
 
-export async function subscribeForOffers (): Promise<void> {
-  if (!libp2p) {
-    throw new Error('Libp2p not initialized')
-  }
-
+export async function subscribeForOffers (libp2p: Libp2p): Promise<void> {
   for (const offer of await Offer.findAll()) {
-    subscribeForOffer(offer)
+    subscribeForOffer(libp2p, offer)
   }
 }
 
-export async function initComms (): Promise<void> {
-  if (libp2p) {
+export async function initComms (app: Application): Promise<void> {
+  if (app.get('libp2p')) {
     throw new Error('libp2p node already spawned')
   }
-  libp2p = await initLibp2p()
+  app.set('libp2p', await initLibp2p())
+  const notificationService = app.service(ServiceAddresses.NOTIFICATION)
+  _messageHandler = messageHandler(notificationService)
 }
 
-export async function stop (): Promise<void> {
+export async function stop (app: Application): Promise<void> {
   for (const [, room] of rooms) {
     room.leave()
```

Comment:
I guess you should remove the room from the `Map` as well right? Just to properly cleanup...

Reply:
Yes, Done